lib/rubocop/target_ruby.rb

```diff
@@ -141,9 +141,17 @@ def find_version
           return versions.compact.min
         end
 
+        return gem_requirement_version(version) if version.send_type?
```

Comment:
Probably we should do a more precise node match here, as there's a slight chance some other code might be used here.

Reply:
Thanks bbatsov!

There is now a more precise match pattern.

Reply:
Looks good! Maybe you can merge it with `if version.array_type?`, also capturing the versions, and you can reuse the same `map`, `compact`, `min`. Maybe not though, just a thought

Reply:
I'll look into it @marcandre!

Reply:
`{ (array $str+) ... } ` can help, and there's https://nodepattern.herokuapp.com/ to help test things.

Reply:
So now the `map`, `compact` and `min` are reused and I definitely think it looks more DRY.

I managed to merge it by creating a `version_from_array` method but didn't do any new `NodePattern`.